app/controllers/concerns/steps/steps_list.rb

```diff
@@ -0,0 +1,43 @@
+module Steps
+  module StepsList
+    extend ActiveSupport::Concern
+
+    included do
+      before_action :steps
+    end
+
+    def change_current_step(step_number)
+      steps.each do |step|
+        if step[:number] == step_number
+          step[:is_current_step] = true
+        end
+      end
+    end
+
+    private
+
+    def steps
+      @steps ||= [
+        {
+          number: 1, title: 'YOUR INFO', is_current_step: false
+        },
+        {
+          number: 2, title: 'SELECT PLAN', is_current_step: false
+        },
+        {
+          number: 3, title: 'ADD-ONS', is_current_step: false
+        },
+        {
+          number: 4, title: 'SUMMARY', is_current_step: false
+        }
+      ]
+    end
+
+    def path_by_step
+      case onboarding.current_step
+      when 2
+        select_plan_onboarding_path(onboarding)
+      end
+    end
```

Comment:
- I don't think it's safe to depend on the fact that there will be an `onboarding` method available on any controller where we include this module. I think this could be much more reusable if you receive the step as an argument and the method just returns the path for each step
- What happens if step is different than 2? Can you add the paths for the other steps on this method?
```suggestion
    def path_by_step(step)
      case step
      when 1
        # ...
      when 2
        select_plan_onboarding_path(onboarding)
      when 3
        # ...
      when 4
        # ...
      else
        raise StandardError, "Step not found"
      end
    end
```

Reply:
I like it, it could be reusable receiving just the step number, ok!